# Membership Calculations

Trying to understand what percentage of PCA members are in the churches with FFO...

Jonathan Barlow (@barlowjon)


## First, Load the Data

In [66]:
# Prepare the libraries
import pandas as pd

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

# Load the two CSV files

# this is the 'save the pca' dataset
# https://www.savethepca.com/downloads/
left = pd.read_csv("stats_left.csv") 

# this is the datafile from the PCA sans headers and footers
# https://presbyteryportal.pcanet.org/Report/StatsReport
right = pd.read_csv("stats_right.csv")

left.describe()

Presbytery                     Church Phone Website Email  \
count                 1964                       1964  1963    1964  1964   
unique                  87                       1365  1830    1734  1777   
top     Mississippi Valley  First Presbyterian Church     -       -     -   
freq                    50                         54   117     229   180   

       Pastor Review Date Elders on Website? (Y/N) Deacons on Website? (Y/N)  \
count    1964        1528                     1528                      1526   
unique   1715          12                        2                         2   
top         -    09-27-25                        Y                         Y   
freq      235         672                     1159                       837   

       Other Staff on Website? (Y/N) Shepherdesses? (Y/N) Deaconesses? (Y/N)  \
count                           1528                 1185                904   
unique                             2                    2                  2   
top                                Y                    N                  N   
freq                             968                 1151                813   

       Website Status  
count            1964  
unique              8  
top        Functional  
freq             1528

## Pull Out Just the Churches with FFO

In [67]:
leftFfo = left[(left["Shepherdesses? (Y/N)"].astype(str).str.upper().str.contains("Y")) |
              (left["Deaconesses? (Y/N)"].astype(str).str.upper().str.contains("Y"))].copy()

leftFfo.describe()

Presbytery                        Church Phone  \
count                     111                           111   111   
unique                     43                           100   103   
top     Metropolitan New York  Redeemer Presbyterian Church     -   
freq                        9                             6     9   

                 Website Email Pastor Review Date Elders on Website? (Y/N)  \
count                111   111    111         111                      111   
unique               111   109    106           9                        2   
top     ascensionphx.org     -      -    09-27-25                        Y   
freq                   1     3      6          40                      109   

       Deacons on Website? (Y/N) Other Staff on Website? (Y/N)  \
count                        111                           111   
unique                         2                             2   
top                            Y                             Y   
freq                          99                           110   

       Shepherdesses? (Y/N) Deaconesses? (Y/N) Website Status  
count                   110                107            111  
unique                    2                  2              1  
top                       N                  Y     Functional  
freq                     76                 91            111

Okay, looks like we have 111 churches with FFO.

## Fix Missing Pastor Names

Since we're going to merge on pastor and church name, we need to be able to match like to like in the two datasets. The FFO file had a dash for missing pastors.

In [68]:
right['Pastor'] = right['Pastor'].fillna('-').astype(str).str.strip()
right.loc[right['Pastor'] == '', 'Pastor'] = '-'

# Set Calvary Church's Pastor to Match FFO Data
right.loc[
    (right['City'] == 'Willow Grove') &
    (right['St'] == 'PA') &
    (right['Church'] == 'Calvary Presbyterian Church'),
    'Pastor'
] = 'Rev. Angel Gomez'


# Redeemer (has three pastors listed in the pca stats sheet)

right.loc[
    (right['City'] == 'Winston-Salem') &
    (right['St'] == 'NC') &
    (right['Church'] == 'Redeemer Presbyterian Church'),
    'Pastor'
] = 'Rev. Giorgio W.K. Hiatt'


# King's Cross Flushing - FFO data had the wrong pastor

leftFfo.loc[
    (leftFfo['Presbytery'] == 'Metropolitan New York') &
    (leftFfo['Church'] == "King's Cross Church"),
    'Pastor'
] = 'James Park'

# Redeemer San Antonio

right.loc[
    (right['City'] == 'San Antonio') &
    (right['St'] == 'TX') &
    (right['Church'] == 'Redeemer Presbyterian Church'),
    'Pastor'
] = 'Rev. James Paul Hahn Jr.'


# Christ the King Dorchester has a data quality problem (extra space in name) in the pca stats data

right.loc[
    (right['City'] == 'Dorchester') &
    (right['St'] == 'MA') &
    (right['Church'] == 'Christ the King  - Dorchester'),
    'Church'
] = 'Christ the King - Dorchester'


## Merge the PCA Stats into the Data

In [69]:
# Normalize the Church name and Pastor columns for better matching
def normalize_church(s):
    return str(s).strip().lower()

def normalize_pastor(s):
    s = str(s).strip().lower()
    parts = s.split()
    return parts[-1] if parts else ''

leftFfo['Church_norm'] = leftFfo['Church'].apply(normalize_church)
leftFfo['Pastor_norm'] = leftFfo['Pastor'].apply(normalize_pastor)
right['Church_norm'] = right['Church'].apply(normalize_church)
right['Pastor_norm'] = right['Pastor'].apply(normalize_pastor)

# Merge the right sheet into the left sheet, keeping all rows from left
merged = pd.merge(leftFfo, right, on=['Church_norm', 'Pastor_norm'], how='left', suffixes=('_left', '_right'))

merged

Presbytery  \
0                    Arizona   
1                    Arizona   
2                 Blue Ridge   
3                 Blue Ridge   
4                    Calvary   
5                    Calvary   
6                Canada West   
7             Catawba Valley   
8             Catawba Valley   
9           Central Carolina   
10          Central Carolina   
11          Central Carolina   
12          Central Carolina   
13           Central Florida   
14           Central Florida   
15           Central Indiana   
16                Chesapeake   
17                Chesapeake   
18                Chesapeake   
19                Chesapeake   
20             Chicago Metro   
21            Eastern Canada   
22            Eastern Canada   
23          Eastern Carolina   
24          Eastern Carolina   
25          Eastern Carolina   
26          Eastern Carolina   
27          Eastern Carolina   
28          Eastern Carolina   
29      Eastern Pennsylvania   
30      Eastern Pennsylvania   
31      Eastern Pennsylvania   
32               Great Lakes   
33                  Heritage   
34                  Heritage   
35                 Highlands   
36                 Highlands   
37          Hills and Plains   
38             Houston Metro   
39             Houston Metro   
40             Houston Metro   
41             Houston Metro   
42               James River   
43               James River   
44             Metro Atlanta   
45             Metro Atlanta   
46             Metro Atlanta   
47             Metro Atlanta   
48             Metro Atlanta   
49     Metropolitan New York   
50     Metropolitan New York   
51     Metropolitan New York   
52     Metropolitan New York   
53     Metropolitan New York   
54     Metropolitan New York   
55     Metropolitan New York   
56     Metropolitan New York   
57     Metropolitan New York   
58                  Missouri   
59                  Missouri   
60                  Missouri   
61                 Nashville   
62            New York State   
63            New York State   
64               North Texas   
65       Northern California   
66               Ohio Valley   
67               Ohio Valley   
68               Ohio Valley   
69              Philadelphia   
70              Philadelphia   
71              Philadelphia   
72              Philadelphia   
73   Philadelphia Metro West   
74   Philadelphia Metro West   
75            Piedmont Triad   
76                Pittsburgh   
77             Platte Valley   
78             Platte Valley   
79                   Potomac   
80                   Potomac   
81                   Potomac   
82                   Potomac   
83            Rocky Mountain   
84            Rocky Mountain   
85            Rocky Mountain   
86            Rocky Mountain   
87               South Coast   
88               South Coast   
89               South Coast   
90               South Coast   
91               South Coast   
92               South Coast   
93               South Texas   
94               South Texas   
95               South Texas   
96               South Texas   
97      Southern New England   
98      Southern New England   
99      Southern New England   
100     Southern New England   
101        Southwest Florida   
102       Susquehanna Valley   
103         Tennessee Valley   
104         Tennessee Valley   
105         Tennessee Valley   
106         Tennessee Valley   
107                Tidewater   
108              West Hudson   
109              West Hudson   
110              West Hudson   
111                Wisconsin   

                                           Church_left         Phone  \
0                          Ascension Church of Phoenix  601-209-5609   
1                                    New Valley Church  480-940-5560   
2                   Hope Presbyterian Church of Crozet  434-829-2181   
3                          Trinity Presbyterian Church  434-977-3700   
4                         Downtown Presby

# Calculate the Percentages of Total Churchgoers Under FFO Churches

In [78]:

merged['Total'] = pd.to_numeric(merged['Total'].astype(str).str.replace(',', '').str.strip(), errors='coerce')
right['Total'] = pd.to_numeric(right['Total'].astype(str).str.replace(',', '').str.strip(), errors='coerce')

merged['Comm'] = pd.to_numeric(merged['Comm'].astype(str).str.replace(',', '').str.strip(), errors='coerce')
right['Comm'] = pd.to_numeric(right['Comm'].astype(str).str.replace(',', '').str.strip(), errors='coerce')


# calc for churchgoers (total of comm, non-comm)
totalUnderFfo = merged['Total'].sum()
totalGoers = right['Total'].sum()

percentUnderFfo = (totalUnderFfo / totalGoers)*100

print(totalUnderFfo," attendees, representing at least ",percentUnderFfo,"% of PCA Attendees are under FFO")

45070.0  attendees, representing at least  11.41021324212589 % of PCA Attendees are under FFO


In [80]:
# total for communing

totalCommUnderFfo = merged['Comm'].sum()
totalComm = right['Comm'].sum()

percentCommUnderFfo = (totalCommUnderFfo / totalComm)*100

print(totalCommUnderFfo," communing members, representing at least ",percentCommUnderFfo,"% of communing PCA Members are under FFO")

34998.0  communing members, representing at least  11.191982245303896 % of communing PCA Members are under FFO
